ngraph/frontend/frontend_manager/include/frontend_manager/parameters.hpp

```diff
@@ -0,0 +1,119 @@
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::std::istringstream*", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+
+#if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+template <>
+class FRONTEND_API VariantWrapper<std::wstring> : public VariantImpl<std::wstring> {
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::std::wstring", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+#endif
+
+class FRONTEND_API ModelPath {
+    std::string m_model_path;
+
+public:
+    ModelPath(std::string model_path) : m_model_path(std::move(model_path)) {}
+
+    std::string get_model_path() const {
+        return m_model_path;
+    }
+};
+
+class FRONTEND_API WeightsPath {
+    std::string m_weights_path;
+
+public:
+    WeightsPath(std::string weights_path) : m_weights_path(std::move(weights_path)) {}
+
+    std::string get_weights_path() const {
+        return m_weights_path;
+    }
+};
+
+template <>
+class FRONTEND_API VariantWrapper<ModelPath> : public VariantImpl<ModelPath> {
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::ModelPath", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+
+template <>
+class FRONTEND_API VariantWrapper<WeightsPath> : public VariantImpl<WeightsPath> {
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::WeightsPath", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+
+template <>
+class FRONTEND_API VariantWrapper<std::shared_ptr<ngraph::op::Constant>>
+    : public VariantImpl<std::shared_ptr<ngraph::op::Constant>> {
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::Constant", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+
+using ConstantVariant = VariantWrapper<std::shared_ptr<ngraph::op::Constant>>;
+
+template <>
+class FRONTEND_API VariantWrapper<std::map<std::string, ngraph::OpSet>>
+    : public VariantImpl<std::map<std::string, ngraph::OpSet>> {
+public:
+    static constexpr VariantTypeInfo type_info{"Variant::Extensions", 0};
+    const VariantTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+};
+
+using Extensions = std::map<std::string, ngraph::OpSet>;
```

Comment:
@ilyachur is it really extensions?

Reply:
It is not extensions. It is just a map of operation sets. In new Extension mechanism, we have a plan to add information about opset to operation.
It means that we won't add extensions with such API. frontends should have `add_extension` mechanism.

Reply:
So let's keep it temporary here and once new extensions API is ready we will replace this code.